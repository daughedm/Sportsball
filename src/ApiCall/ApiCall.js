import apiKey from './apiKey';
import yesterdaysDate from '../Helpers/dataCleaner'

export const gameSummaryFetch = async (selectedTeam) => {
  const url = `http://api.sportradar.us/mlb/trial/v6.5/en/games/2018/05/23/summary.json?api_key=${apiKey}`
  const response = await fetch()
}
