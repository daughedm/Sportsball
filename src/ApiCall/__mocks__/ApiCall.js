import * as MOCKDATA from './../../mockData/mockdata';

export const gameSummariesFetch = jest.fn().mockImplementation(() => 
  Promise.resolve(MOCKDATA.mockSummaryData)
); 

export const boxScoresFetch = jest.fn().mockImplementation(() =>
  Promise.resolve(MOCKDATA.mockBoxScoreData)
);

export const teamStatsFetch = jest.fn().mockImplementation(() =>
  Promise.resolve(MOCKDATA.mockTeamStatsData)
);