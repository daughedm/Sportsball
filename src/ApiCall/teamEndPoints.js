const teamEndPoints = (team) => {
  switch (team) {
    case 'Blue Jays':
      return '1d678440-b4b1-4954-9b39-70afb3ebbcfa';
     case 'Rays':
       return 'bdc11650-6f74-49c4-875e-778aeb7632d9';
    case 'Orioles':
      return '75729d34-bca7-4a0f-b3df-6f26c6ad3719';
    case 'Red Sox':
      return '93941372-eb4c-4c40-aced-fe3267174393';
    case 'Yankees':
      return 'a09ec676-f887-43dc-bbb3-cf4bbaee9a18';
    case 'Royals':
      return '833a51a9-0d84-410f-bd77-da08c3e5e26e';
    case 'Indians':
      return '80715d0d-0d2a-450f-a970-1b9a3b18c7e7';
    case 'White Sox':
      return '47f490cd-2f58-4ef7-9dfd-2ad6ba6c1ae8';
    case 'Tigers':
      return '575c19b7-4052-41c2-9f0a-1c5813d02f99';
    case 'Twins':
      return 'aa34e0ed-f342-4ec6-b774-c79b47b60e2d';
    case 'Mariners':
      return '43a39081-52b4-4f93-ad29-da7f329ea960';
    case 'Angels':
      return '4f735188-37c8-473d-ae32-1f7e34ccf892';
    case 'Athletics':
      return '27a59d3b-ff7c-48ea-b016-4798f560f5e1';
    case 'Rangers':
      return 'd99f919b-1534-4516-8e8a-9cd106c6d8cd';
    case 'Astros':
      return 'eb21dadd-8f10-4095-8bf3-dfb3b779f107';
    case 'Reds':
      return 'c874a065-c115-4e7d-b0f0-235584fb0e6f';
    case 'Pirates':
      return '481dfe7e-5dab-46ab-a49f-9dcc2b6e2cfd';
    case 'Cubs':
      return '55714da8-fcaf-4574-8443-59bfb511a524';
    case 'Cardinals':
      return '44671792-dc02-4fdd-a5ad-f5f17edaa9d7';
    case 'Brewers':
      return 'dcfd5266-00ce-442c-bc09-264cd20cf455';
    case 'Dodgers':
      return 'ef64da7f-cfaf-4300-87b0-9313386b977c';
    case 'Diamondbacks':
      return '25507be1-6a68-4267-bd82-e097d94b359b';
    case 'Giants':
      return 'a7723160-10b7-4277-a309-d8dd95a8ae65';
    case 'Rockies':
      return '29dd9a87-5bcc-4774-80c3-7f50d985068b';
    case 'Padres':
      return 'd52d5339-cbdd-43f3-9dfa-a42fd588b9a3';
    case 'Phillies':
      return '2142e1ba-3b40-445c-b8bb-f1f8b1054220';
    case 'Mets':
      return 'f246a5e5-afdb-479c-9aaa-c68beeda7af6';
    case 'Marlins':
      return '03556285-bdbb-4576-a06d-42f71f46ddc5';
    case 'Braves':
      return '12079497-e414-450a-8bf2-29f91de646bf';
    case 'Nationals':
      return 'd89bed32-3aee-4407-99e3-4103641b999a';
    default:
      return '80715d0d-0d2a-450f-a970-1b9a3b18c7e7';
  }
}