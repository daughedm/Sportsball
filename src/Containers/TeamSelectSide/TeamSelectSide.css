.unordered-list {
  list-style-type: none;
}

.scrollable-menu {
  height: auto;
  max-height: 170px;
  overflow-x: hidden;
  width: 100%;
  border-radius: 4px;
  border: 1px solid #3C3C3A;
  margin-top: 25px;
  background-color: #F4F1E7;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);
  padding: 0 20px 0 20px;
}

.side-form {
  width: 100%;
  background-color: rgba(60, 60, 58, 0.18);
  border: 1px solid #F4F1E7;
  border-radius: 4px;
  margin: 60px auto;
  padding: 35px;
}

.form-header {
  color: #F26F44;
  font-family: "Noto Sans";
  font-size: 25px;
  font-weight: 700;
  margin-top: 0;
  margin-bottom: 20px;
  letter-spacing: 1px;
}

.drop-down-button {
  display: block;
  width: 100%;
  height: 50px;
  border-radius: 4px;
  border: 1px solid #3C3C3A;
  margin-top: 25px;
  background-color: #F4F1E7;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);
  padding: 10px;
  font-size: 14px;
  color: #3C3C3A;
}

.submit-button {
  display: block;
  margin-top: 25px;
  width: 100%;
  height: 50px;
  border: none;
  border-radius: 4px;
  background: linear-gradient(0deg, #F26F44 11%, #F98059 52%, #FF8F6B 100%);
  color: #F4F1E7;
  font-size: 16px;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);
  cursor: pointer;
}

.anchor-toggle {
  font-size: 14px;
  display: block;
  text-decoration: underline;
  color: #F4F1E7;
  font-style: italic;
  padding-top: 12px;
  cursor: pointer;
}

li a {
  text-decoration: none;
  font-size: 16px;
  color: #3C3C3A;
 }

.list-item {
 padding: 15px 0 15px 0;
 height: 50px;
 border-bottom: 1px solid #808080;
}

.unordered-list :hover {
  background-color: #ffe4db;
}

::-webkit-scrollbar {
  width: 8px;
  height: 8px;
}

::-webkit-scrollbar-button {
  width: 0px;
  height: 0px;
}

::-webkit-scrollbar-thumb {
  background: #f26f44;
  border: 2px solid transparent;
  border-radius: 44px;
}

::-webkit-scrollbar-thumb:hover {
  background: #f26f44;
}

::-webkit-scrollbar-thumb:active {
  background: #f26f44;
}

::-webkit-scrollbar-track {
  background: #f4f1e7;
  border: 45px none #ffffff;
  border-radius: 44px;
}

::-webkit-scrollbar-track:hover {
  background: transparent;
}

::-webkit-scrollbar-track:active {
  background: transparent;
}

::-webkit-scrollbar-corner {
  background: transparent;
}