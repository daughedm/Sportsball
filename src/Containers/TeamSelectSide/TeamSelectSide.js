import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { findGame, teamStatCleaner } from '../../Helpers/dataCleaner'
import { addSelectedTeamToStore } from '../../Actions/userActions';
import './TeamSelectSide.css';
import { addSingleSummaryToStore, addSingleBoxScoreToStore, addTeamStatsToStore } from '../../Actions';
import { teamStatsFetch } from '../../ApiCall/ApiCall';
import TeamScroll from './../TeamScroll/TeamScroll';

export class TeamSelectSide extends Component {

  render() {
    return ( 
      <form
        className = "side-form"
        type = "submit"
      >
        <h3 className = "form-header" > Switch Teams </h3> 
          <TeamScroll />
      </form>
    )
  }
}

TeamSelectSide.propTypes = {

}

export default TeamSelectSide
