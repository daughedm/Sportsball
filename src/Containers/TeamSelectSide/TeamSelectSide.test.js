import {TeamSelectSide, mapDispatchToProps, mapStateToProps} from './TeamSelectSide';
import React from 'react';
import { shallow } from 'enzyme';
import { teamStatsFetch } from './../../ApiCall/ApiCall';


jest.mock('./../../apiCall/apiCall');

describe('TeamSelectSide', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<TeamSelectSide/>);
  });
  
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});