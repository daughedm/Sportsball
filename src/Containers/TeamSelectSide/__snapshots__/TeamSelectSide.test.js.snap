// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamSelectSide should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TeamSelectSide />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="form-header"
>
           Switch Teams 
</h3>,
        <Connect(TeamScroll) />,
      ],
      "className": "side-form",
      "type": "submit",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Switch Teams ",
          "className": "form-header",
        },
        "ref": null,
        "rendered": " Switch Teams ",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="form-header"
>
             Switch Teams 
</h3>,
          <Connect(TeamScroll) />,
        ],
        "className": "side-form",
        "type": "submit",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Switch Teams ",
            "className": "form-header",
          },
          "ref": null,
          "rendered": " Switch Teams ",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
